app/controllers/api/v1/profiles_controller.rb

```diff
@@ -0,0 +1,61 @@
+class Api::V1::ProfilesController < Api::V1::ApplicationController
+  before_action :authenticate_user!
+
+  def new
+    if user_signed_in?
+      @profile = current_user.build_profile(profile_params)
+      
+      if @profile.save
+        render json: @profile, status: :created
+      else
+        render json: { errors: @profile.errors.full_messages }, status: :unprocessable_entity
+      end
+    else
+      render json: { message: 'Authentication error' }, status: :unauthorized
+    end
+  end
+
+  def show
```

Comment:
`show` & `edit` are the same logic. Also, you have `update`, so `edit` seems like it's not needed. And you can make this a 1liner: `render json: current_user.profile`